tests/dialects/test_mpi.py

```diff
@@ -13,46 +14,70 @@
 def test_mpi_combo():
     printer = Printer(target=Printer.Target.MLIR)
 
-    region0 = Region.from_operation_list([
-        memref := Alloc.get(f64, 32, [100, 14, 14]),
-        dest := Constant.from_int_and_width(1, t_int),
-        req := ISend.get(memref, dest, 1),
-        res := IRecv.get(dest, memref.results[0].typ, 1),
-        test_res := Test.get(res.request),
-        tag := GetStatusField.get(test_res.status, StatusTypeField.MPI_TAG),
-        src := GetStatusField.get(test_res.status, StatusTypeField.MPI_SOURCE),
-        err := GetStatusField.get(test_res.status, StatusTypeField.MPI_ERROR),
-        code2 := Wait.get(res.request)
+    module = ModuleOp.from_region_or_ops([
+        func.FuncOp.from_callable('main', [], [], lambda: [
+            mpi.Init.build(),
+            rank := mpi.CommRank.get(),
+            lit0 := arith.Constant.from_int_and_width(0, 32),
+            cond := arith.Cmpi.from_mnemonic(rank, lit0, 'eq'),
+            buff := memref.Alloc.get(f64, 32, [100, 14, 14]),
+            scf.If.get(cond, [], [  # if rank == 0
+                dest := arith.Constant.from_int_and_width(1, mpi.t_int),
+                mpi.Send.get(buff, dest, 1),
+                # mpi.Wait.get(req, ignore_status=False),
+                scf.Yield.get(),
+            ], [  # else
+                source := arith.Constant.from_int_and_width(0, mpi.t_int),
+                status := mpi.Recv.get(source, buff, 1, ignore_status=False),
+                GetStatusField.get(status, StatusTypeField.MPI_TAG),
+                #mpi.Wait.get(recv.request),
+                scf.Yield.get(),
+            ]),
+            mpi.Finalize.build(),
+            func.Return.get()
+        ])
     ])  # yapf: disable
 
     expected = \
         """"builtin.module"() ({
-  %0 = "memref.alloc"() {"alignment" = 32 : i64, "operand_segment_sizes" = array<i32: 0, 0>} : () -> memref<100x14x14xf64>
-  %1 = "arith.constant"() {"value" = 1 : si32} : () -> si32
-  %2 = "mpi.isend"(%0, %1) {"tag" = 1 : si32} : (memref<100x14x14xf64>, si32) -> #mpi.request
-  %3, %4 = "mpi.irecv"(%1) {"tag" = 1 : si32} : (si32) -> (memref<100x14x14xf64>, #mpi.request)
-  %5, %6 = "mpi.test"(%4) : (#mpi.request) -> (i1, #mpi.status)
-  %7 = "mpi.status.get"(%6) {"field" = "MPI_TAG"} : (#mpi.status) -> si32
-  %8 = "mpi.status.get"(%6) {"field" = "MPI_SOURCE"} : (#mpi.status) -> si32
-  %9 = "mpi.status.get"(%6) {"field" = "MPI_ERROR"} : (#mpi.status) -> si32
-  %10 = "mpi.wait"(%4) : (#mpi.request) -> si32
+  "func.func"() ({
```

Comment:
This string is getting to be quite long, and feels more and more like an integration test as opposed to a unit test. I'm not sure what is being tested here

Reply:
Strong +1